src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaTransportContext.cs

```diff
@@ -0,0 +1,128 @@
+namespace MassTransit.KafkaIntegration;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+using MassTransit.Configuration;
+using Serializers;
+using Transports;
+
+
+public class KafkaTransportContext<TKey, TValue> :
+    BaseSendTransportContext,
+    KafkaSendTransportContext<TKey, TValue>
+    where TValue : class
+{
+    readonly IHeadersSerializer _headersSerializer;
+    readonly IHostConfiguration _hostConfiguration;
+    readonly IAsyncSerializer<TKey> _keySerializer;
+    readonly ISendPipe _sendPipe;
+    readonly IProducerContextSupervisor _supervisor;
+    readonly KafkaTopicAddress _topicAddress;
+    readonly IAsyncSerializer<TValue> _valueSerializer;
+
+    public KafkaTransportContext(IHostConfiguration hostConfiguration, string topicName,
+        IProducerContextSupervisor supervisor, IHeadersSerializer headersSerializer, IAsyncSerializer<TKey> keySerializer,
+        IAsyncSerializer<TValue> valueSerializer, Action<ISendPipeConfigurator> configureSend, ISerialization serialization)
+        : base(hostConfiguration, serialization)
+    {
+        var sendConfiguration = new SendPipeConfiguration(hostConfiguration.Topology.SendTopology);
+        configureSend?.Invoke(sendConfiguration.Configurator);
+
+        _hostConfiguration = hostConfiguration;
+        _supervisor = supervisor;
+        _sendPipe = sendConfiguration.CreatePipe();
+        _topicAddress = new KafkaTopicAddress(hostConfiguration.HostAddress, topicName);
+        _headersSerializer = headersSerializer;
+        _keySerializer = keySerializer;
+        _valueSerializer = valueSerializer;
+    }
+
+    public override IEnumerable<IAgent> GetAgentHandles()
```

Comment:
So you use it here, did you mean to use it in Event Hub too?

Reply:
yes, I'm using in both Riders, I wanna stop producer supervisor when Producer is removed from cache.